src/ProxyManager/GeneratorStrategy/FileWriterGeneratorStrategy.php

```diff
@@ -74,9 +75,12 @@ public function generate(ClassGenerator $classGenerator) : string
      */
     private function writeFile(string $source, string $location) : void
     {
-        $tmpFileName = tempnam($location, 'temporaryProxyManagerFile');
+        $directory   = dirname($location);
```

Comment:
In this case, we need to check if the `$location` is a directory first, otherwise, we will move up one directory.

Reply:
`$location` cannot be directory because it's value returned from `FileLocatorInterface::locate()`, which is supposed to return "full path for a given file name".

Reply:
See https://github.com/Ocramius/ProxyManager/pull/438/files/67ddd7b00c70f40cb8469a6bf4e474da44189e95#diff-f7ba53e97e34edc37d8bc3477bea48f2R61